### DISCUSSION TITLE
Enhancement: Celery integration

### DISCUSSION THREAD

**Discussion by giulioindev** [litestar-org/litestar#3653] [Q&A] [answered]
### Summary

It would be great if the documentation could describe how Celery can be integrated with a Litestar project.

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3652">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3652/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3652/pledge.svg">
</picture>
</a>

**Comment by provinzkraut** [comment, marked as answer]
What exactly are you looking for to be documented? What kind of integration are you thinking about? Are there concrete question you'd like to have answered?

In general, there isn't anything special you need to do to "integrate" celery with Litestar. You can pretty much just set up your celery app and start dispatching tasks to it from your Litestar app just like you would with every other application.

**Reply by giulioindev** [reply]
So if I got it there is nothing special to account for (as must be done when using Django for instance), just set Celery up aside from the Litestar app and you are ready to go, right?

**Reply by provinzkraut** [reply]
Pretty much, yes.

**Reply by giulioindev** [reply]
Great, thanks a lot!

**Reply by ddahan** [reply]
@giulioindev after installing Celery with FastAPI in 3 lines, I realized than Django specific installation was more due to Django specificities, than Django providing an integration themselves. Basically, a setup of Celery consists on 3 steps:

```
# 1: creates a Celery instance
celery_app = Celery(__name__) 

# 2: get config from somewhere (in my case, Pydantic BaseSettings object)
celery_app.config_from_object(obj=settings.model_dump(), namespace="celery")

# 3: guide Celery to find your tasks automatically
celery_app.autodiscover_tasks(packages=["app.tasks"])
```

And basically, that's it.
